### DISCUSSION TITLE
Configuring AGH for DOT

### DISCUSSION THREAD

**Discussion by ric-m** [AdguardTeam/AdGuardHome#7372] [General]
DOT implementation would be one huge advantage over my current installations using PiHole. I cannot find a simple tutorial outlining the considerations and steps to implement DOT. The Wiki talks about this but in the context of a cloud server; however, I'm using a Raspberry Pi (64bit Debian Bookworm) and docker containers. There's a section on docker but not on DOT _and_ docker.
This link:
https://www.snbforums.com/threads/dot-settings-reset.83047/
seems to suggest that DOT will work pretty much out of the box without any additional SSL certs. The AGH Wiki also mentions DOT working "out of the box" but in the context of the required _protocols_. Additional configuration (eg, domain, certs, etc) isn't mentioned.
Can anyone clarify?
Thanks,
Ric

**Comment by Dynamic5912** [comment]
No additional config is needed. 

Just add the required DOH/DOT upstream DNS servers and away you go :)

**Reply by ric-m** [reply]
Thank you for that. I have been testing Adguard Home again after earlier attempts failed to resolve any queries. I have tracked (I think) the problem to quite a different issue. 
Response times seem quite good (11-20ms) and the provided list of known DNS providers is extremely useful.

I have an entry for "https://dns10.quad9.net:443/dns-query" which isn't in any of my manually-entered lists and I'm wondering where that came from. It's not in the router and not in the Raspberry Pi Network configs, so it's coming from somewhere else... I shall check to see if AGH has its own configuration file somewhere..

So far, very impressed.

**Reply by Dynamic5912** [reply]
> Thank you for that. I have been testing Adguard Home again after earlier attempts failed to resolve any queries. I have tracked (I think) the problem to quite a different issue. 
> 
> Response times seem quite good (11-20ms) and the provided list of known DNS providers is extremely useful.
> 
> 
> 
> I have an entry for "https://dns10.quad9.net:443/dns-query" which isn't in any of my manually-entered lists and I'm wondering where that came from. It's not in the router and not in the Raspberry Pi Network configs, so it's coming from somewhere else... I shall check to see if AGH has its own configuration file somewhere..
> 
> 
> 
> So far, very impressed.

Where are you seeing that request?

**Reply by ric-m** [reply]
> Where are you seeing that request?

It's in the list of "Top Upstreams" and "Average Upstream Response Time".

I had a thought a few moments ago that both entries may be legacy values from when Adguard was first installed because both look back over 24 hours... In which case the entries should disappear some time later today.

Fastest response times are 10, 11, 11 and 12ms for the top 4. Which is great.
